src/ILCompiler.Compiler/src/Compiler/RawMainMethodRootProvider.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler
+{
+    public class RawMainMethodRootProvider : ICompilationRootProvider
```

Comment:
Should this be in the Wasm specific assembly? This is obviously not useful for any other IL compiler. Wasm only has it because it can't be used for anything right now either.

Reply:
Done (though I could see it being useful for future bootstrapping projects)